packages/block-library/src/navigation-link/index.php

```diff
@@ -367,5 +387,42 @@ function register_block_core_navigation_link() {
 			'variations'      => array_merge( $built_ins, $variations ),
 		)
 	);
+
+	// Register actions for all post types and taxonomies that may not yet be registered.
+	// This needs to happen in this function, because otherwise any post types/taxonomies
+	// registered **before** this block will attempt to register on a block that does not yet exist.
+	add_action( 'registered_post_type', 'register_block_core_navigation_link_post_type_variation', 10, 2 );
+	add_action( 'registered_taxonomy', 'register_block_core_navigation_link_taxonomy_variation', 10, 3 );
 }
 add_action( 'init', 'register_block_core_navigation_link' );
+
+/**
+ * Register custom post type variations for navigation link on post type registration
+ * Handles all post types registered after the block is registered in register_navigation_link_post_type_variations
+ *
+ * @param string       $post_type The post type name passed from registered_post_type filter.
+ * @param WP_Post_Type $post_type_object The post type object passed from registered_post_type.
+ * @return void
+ */
+function register_block_core_navigation_link_post_type_variation( $post_type, $post_type_object ) {
```

Comment:
```suggestion
function gutenberg_register_block_core_navigation_link_post_type_variation( $post_type, $post_type_object ) {
```

As per https://github.com/WordPress/gutenberg/blob/trunk/lib/README.md. Let's wait for confirmation before updating.

Reply:
Ignore. I was incorrect.